### PR TITLE
Bash exe

### PR DESCRIPTION
**Pull request by alexedelsburg**
Begins to implement #754 

**DO NOT MERGE** - I didn't install ruby on the Frankenstein setup yet; I just wanted to get the PR out in the open.

I made the edits and submitted the PR from Windows with git-town commands as I made them work.

### CODE REVIEW

**Comment by alexedelsburg** [comment]
Hmm, circle failed 😦 , but my fork runs fine on my mac. I'll see about getting the full environment setup soon.

File: src/helpers/browser_helpers.sh [hunk 1]
```diff
@@ -7,6 +7,11 @@
 function open_browser {
   local url=$1
 
+  if [[ "$(uname)" == *"MINGW"* ]]; then
```
Comment (kevgo) [inline src/helpers/browser_helpers.sh]: Why the asterisks around `MINGW`?

**Comment by HangingClowns** [comment]
@alexedelsburg @kevgo hey guys what's the holdup? Anything I can do? I'm in branch hell at the office! Anything I can do to help? :)

**Comment by kevgo** [comment]
I'm fine with this PR. Not sure how to test that it runs on Windows though... @charlierudolph any feedback on this so that we can get closer to merging this in?

**Comment by HangingClowns** [comment]
@kevgo if we get some install instructions for windows, I can give it a spin.

**Comment by kevgo** [comment]
@HangingClowns do the manual installation instructions at http://www.git-town.com/install.html work for you? 

- clone the repo to your machine
- add `<code repo>/src` to your PATH

Sorry we don't have a Windows installer yet, but that won't help you anyways since you want to develop on it.

**Comment by HangingClowns** [comment]
funky problems:

```
user@PC MINGW32 ~/PycharmProjects/Project (feature/feature1)
$ git town alias false

user@PC MINGW32 ~/PycharmProjects/Project (feature/feature1)
$ git town config
Main branch:
  master

Perennial branches:
  [none]

Branch Ancestry:
  master

Pull branch strategy:
  rebase

git-hack push flag:
  true

user@PC MINGW32 ~/PycharmProjects/Project (feature/feature1)
$ git sync
git: 'sync' is not a git command. See 'git --help'.

Did you mean one of these?
        svn
        town-sync

user@PC MINGW32 ~/PycharmProjects/Project (feature/feature1)
$ git town sync
error: unsupported subcommand 'sync'
usage: git town
   or: git town alias (true | false)
   or: git town config [--reset | --setup]
   or: git town hack-push-flag [(true | false)]
   or: git town help
   or: git town install-fish-autocompletion
   or: git town main-branch [<branch_name>]
   or: git town perennial-branches [(--add | --remove) <branch_name>]
   or: git town pull-branch-strategy [(rebase | merge)]
   or: git town version

user@PC MINGW32 ~/PycharmProjects/Project (feature/feature1)
$ git town-sync

[feature/add-docs] git fetch --prune
From ssh://bitbucket:7999/ut/auto_index
 x [deleted]         (none)     -> origin/feature/feature2

Feature branches can be branched directly off
master or from other feature branches.

The former allows to develop and ship features completely independent of each other.
The latter allows to build on top of currently unshipped features.

  1: master
  2: feature/feature1

Please specify the parent branch of feature/add-docs by name or number (default: master): master

/c/Users/user/IdeaProjects/git-town/src/helpers/script_helpers.sh: line 139: /tmp/C:\Users\user\IdeaProjects\git-town\src\git-town-sync_C:_Users_user_PycharmProjects_Project : No such file or directory
```

**Comment by alexedelsburg** [comment]
Sorry folks! I completely missed the notifications for this...

If @HangingClowns can get it working, that's a good sign. I don't think Circle supports Windows, so we could have a problem ensuring it continues to work going forward.

File: src/helpers/browser_helpers.sh (hunk 1 above)
Comment (alexedelsburg) [inline src/helpers/browser_helpers.sh]: Wildcard. `uname` on `bash.exe` returns `MINGW64_NT-10.0`

**Comment by kevgo** [comment]
I am using https://www.appveyor.com for Windows testing on a number of other projects, and can set up an instance for Git Town as well! It would be great to get more official support for Windows!

File: src/helpers/browser_helpers.sh (hunk 1 above)
Comment (kevgo) [inline src/helpers/browser_helpers.sh]: Thanks, makes sense!

File: src/helpers/helpers.sh [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 current_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-export PROGRAM="$(echo "$0" | grep -o "[^/]*$")"
+export PROGRAM="$(echo "$0" | sed 's/\\/\//g' | sed 's/://g'| grep -o "[^/]*$")"
```
Comment (charlierudolph) [inline src/helpers/helpers.sh]: Can we extract the logic that is converting windows styles paths into unix styles paths into a function so the logic only lives in one place?

**Comment by kevgo** [comment]
This PR is obsolete with the new [rewrite in Go](https://github.com/Originate/git-town/releases/tag/v4.0.0). Closing this. Please report Windows issues for the new architecture. Thanks!

**Comment by alexedelsburg** [comment]
Apologies for not getting back to this. I finally had time and was pleasantly surprised to learn about the rewrite! I have the binary installed on Windows now and it's working great 👍

**Pull request closed without merging**